softwerkskammer/lib/mailsender/mailsenderService.js

```diff
@@ -38,6 +40,41 @@ function activityMarkdown(activity, language) {
   return markdown;
 }
 
+function createChunkedSendingReportMessage(statusmessages, subject, sender) {
+  const anySendingError = R.any(statusmessage.isErrorMessage, statusmessages);
+  const resultMessage = new Message(
+    {
+      subject,
+      markdown: anySendingError
+        ? `Fehler: ${statusmessages.map((s) => s.contents().additionalArguments.err).join(" ")}`
+        : "E-Mails erfolgreich versendet",
+    },
+    sender,
+  );
+  resultMessage.setTo(sender.email());
+  return resultMessage;
+}
+
+function sendMailInChunks(maxMailSendingChunkSize, allMembers, message, type, sender) {
```

Comment:
Kannst Du das in async/await umwandeln und das statushandling geradeziehen? - Fehler oder Success zurückgeben

Reply:
Die Umsetzung davon hängt glaube ich auch an er Frage, welche der beiden Optionen (s. oben) umgesetzt wird.
Wenn man die Mails im Hintergrund versendet, dann ist es meiner Meinung nach mit `Promise.then` einfacher bzw. dann kann die `sendInChunks` zwar `await` verwenden, aber der Aufrufer von `sendInChunks` darf nicht `await` verwenden.
Bei der zweiten Option ist `await` auf jeden Fall die schönere Variante.

Reply:
passt